<launch>
	<!-- Global params -->
	<rosparam param="image_resize_width">115</rosparam>
	<rosparam param="image_resize_height">44</rosparam>
	<rosparam param="patch_size">(9,9)</rosparam>
	<rosparam param="calibration_file_left" subst_value="True">$(find teach_repeat)/calibration/left_240_miro_b.yaml</rosparam>
	<rosparam param="calibration_file_right" subst_value="True">$(find teach_repeat)/calibration/right_240_miro_b.yaml</rosparam>
	<rosparam param="wait_for_ready">true</rosparam>

	<group ns="/miro/teacher">
		<!-- Hold head in place and reset odometry -->
		<node pkg="teach_repeat" type="miro_setup.py" name="miro_setup" output="screen">
			<rosparam param="lift">30</rosparam>
			<!--<rosparam param="yaw">-15</rosparam> Miro1 Have issue with yaw-->
			<rosparam param="yaw">-10</rosparam>
			<rosparam param="pitch">0</rosparam>
			<rosparam param="reset_odom">true</rosparam>
			<rosparam param="disable_cliff_sensors">true</rosparam>
			<rosparam param="trigger_services">ready_data_save,ready_data_collect,ready_teleop</rosparam>

			<remap from='/miro/sensors/odom/reset' to='sensors/odom/reset' />
			<remap from='/miro/control/kinematic_joints/set_fixed_state' to='control/kinematic_joints/set_fixed_state' />
			<remap from='/miro/control/kinematic_joints/fixed/enable' to='control/kinematic_joints/fixed/enable' />
			<remap from='/miro/control/kinematic_joints/fixed/disable' to='control/kinematic_joints/fixed/disable' />
			<remap from="/miro/control/kinematic_joints/at_set_point" to="control/kinematic_joints/at_set_point" />
		</node>

		<!-- Teleop to control Miro -->
		<arg name="speed_lin" value="1.0" />
		<arg name="speed_ang" value="1.0" />
		<!--<include file="$(find teach_repeat)/launch/teleop.launch">
			<arg name="cmd_vel_topic" value="rob01/control/cmd_vel" />
			<arg name="speed_linear" value="$(arg speed_lin)" />
			<arg name="speed_angular" value="$(arg speed_ang)" />
		</include> -->
		<include file="$(find teach_repeat)/launch/teleop_joy_miro_b.launch">
			<arg name="cmd_vel_topic" value="control/cmd_vel" />
			<arg name="speed_linear" value="$(arg speed_lin)" />
			<arg name="speed_angular" value="$(arg speed_ang)" />
			<arg name="wait_for_ready" value="true" />
		</include>

		<!-- Stitch image files for matching -->
		<node pkg="teach_repeat" type="image_stitcher.py" name="image_stitcher" output="screen">
			<rosparam param="half_field_of_view">45</rosparam>
			<rosparam param="half_camera_offset">27</rosparam>
			<rosparam param="extra_pixels_proportion">0.0</rosparam>
			<rosparam param="blank_pixels_proportion">0.36</rosparam>
		</node>

		<!-- Collect image data and append (now only exist in teacher robot file below) -->
		<node pkg="teach_repeat" type="data_collect.py" name="data_collect" output="screen">
			<remap from="odom" to="sensors/odom/integrated" />
			<remap from="image" to="sensors/cam/both" />
			<rosparam param="distance_threshold">0.1</rosparam>
			<rosparam param="angle_threshold_deg">10.0</rosparam>
		</node>

		<!-- Save Images and pose in a directory -->
		<node pkg="teach_repeat" type="data_save.py" name="data_save" output="screen">
			<rosparam param="save_dir">~/miro/data/u_and_left_turn_comparison/teacher_teach_data</rosparam>
			<rosparam param="timestamp_folder">false</rosparam>
		</node>
	</group>
</launch>
